docs/preview/logdecorations.md

```diff
@@ -0,0 +1,260 @@
+
+# Log decorations
+
+
+Task authors should be able to control how the log output is displayed to the end user.
+This outlines different decoration options that are available.
+
+## Special lines
+* Errors
+  * `##[error] I am an error` 
+* Warnings
+  * `##[warning] I am a warning` 
+* Commands
+  * `##[warning] I am a command/a tool` 
+* Sections
+  * `##[section] I am a section, which is usually whole task step. I am not typically used.` 
+
+## Collapse
+
+>This is a draft spec and is subject to change
+
+>Note that that if you log an error using ```##vso[task.logissue]error/warning message``` command (see [logging commands](https://github.com/Microsoft/azure-pipelines-tasks/blob/master/docs/authoring/commands.md) here) we will surface those errors in build view and when clicked , we will automatically jump to that particular line. If it's already part of a group, we will auto-expand the group.
+
+Task authors can mark any part of the log as a collapsible region using these decorations:
+
+Starting the collapsible region - `##[startgroup<optionalGroupName>]`
+
+Ending the collapsible region - `##[endgroup<optionalGroupName>]`
```

Comment:
is name needed for endgroup? is it effectively ignored if specified? 

it's kind of like writing

```
if (some condition)
  …
endif (some condition)
```

Reply:
I am thinking if a group has a name, then end has to have a name. We will ignore if that isn't the case.
So if `startgroup:foo` doesn't have any `endgroup:foo` we will group everything.

Reply:
This will support nested grouping as shown in example

Reply:
i see. are nested groups really needed?